kong/plugins/hmac-auth/access.lua

```diff
@@ -95,82 +94,87 @@ local function validate_params(params, conf)
+
+local function retrieve_hmac_fields(authorization_header)
   local hmac_params = {}
-  local authorization_header = headers[header_name]
+
   -- parse the header to retrieve hamc parameters
   if authorization_header then
-    local iterator, iter_err = ngx_gmatch(authorization_header, "\\s*[Hh]mac\\s*username=\"(.+)\",\\s*algorithm=\"(.+)\",\\s*headers=\"(.+)\",\\s*signature=\"(.+)\"")
+    local iterator, iter_err = re_gmatch(authorization_header,
+                                          "\\s*[Hh]mac\\s*username=\"(.+)\",\\s*algorithm=\"(.+)\",\\s*headers=\"(.+)\",\\s*signature=\"(.+)\"")
     if not iterator then
-      ngx_log(ngx.ERR, iter_err)
+      kong.log.err(iter_err)
       return
     end
 
     local m, err = iterator()
     if err then
-      ngx_log(ngx.ERR, err)
+      kong.log.err(err)
       return
     end
 
     if m and #m >= 4 then
       hmac_params.username = m[1]
       hmac_params.algorithm = m[2]
-      hmac_params.hmac_headers = split(m[3], " ")
+      hmac_params.hmac_headers = stringx.split(m[3], " ")
       hmac_params.signature = m[4]
     end
   end
 
-  if conf.hide_credentials then
-    request.clear_header(header_name)
-  end
-
   return hmac_params
 end
 
+
 -- plugin assumes the request parameters being used for creating
 -- signature by client are not changed by core or any other plugin
-local function create_hash(request, request_uri, hmac_params, headers)
+local function create_hash(request_uri, hmac_params)
   local signing_string = ""
   local hmac_headers = hmac_params.hmac_headers
-  local count = #hmac_headers
 
+  local count = #hmac_headers
   for i = 1, count do
     local header = hmac_headers[i]
-    local header_value = headers[header]
+    local header_value = kong.request.get_header(header)
 
     if not header_value then
       if header == "request-line" then
         -- request-line in hmac headers list
-        local request_line = fmt("%s %s HTTP/%s", ngx.req.get_method(),
-                                 request_uri, ngx.req.http_version())
+        local request_line = fmt("%s %s HTTP/%s", kong.request.get_method(),
+                                 request_uri, kong.request.get_http_version())
         signing_string = signing_string .. request_line
+
       else
         signing_string = signing_string .. header .. ":"
       end
+
     else
       signing_string = signing_string .. header .. ":" .. " " .. header_value
     end
+
     if i < count then
       signing_string = signing_string .. "\n"
     end
   end
+
   return hmac[hmac_params.algorithm](hmac_params.secret, signing_string)
 end
 
-local function validate_signature(request, hmac_params, headers)
-  local signature_1 = create_hash(request, ngx.var.request_uri, hmac_params, headers)
-  local signature_2 = ngx_decode_base64(hmac_params.signature)
+
+local function validate_signature(hmac_params)
+  local signature_1 = create_hash(ngx.var.request_uri, hmac_params)
```

Comment:
It's unfortunate we still need to rely on `ngx.var.request_uri` here, but I guess it's because we don't have an exact equivalent in the PDK, right? Would this be our best shot sticking to the current PDK API?

```lua
local function get_path_and_query()
  local p = ngx.request.get_path()
  local q = ngx.request.get_raw_query()
  return (q ~= "") and (p .. "?" .. q) or p
end
```
(which is inefficient because `get_path` strips the query from `ngx.var.request_uri` and then this adds it back, I know.)

Is this pattern common enough to warrant a `kong.request.get_path_and_query` function? (The aws-lambda plugin does the same as the above, but that's mostly for backwards compat with the "request_uri" info being forwarded)